sklearn/isotonic.py

```diff
@@ -325,7 +325,7 @@ def fit(self, X, y, sample_weight=None):
         new input data.
         """
         check_params = dict(accept_sparse=False, ensure_2d=False,
-                            dtype=[np.float64, np.float32])
+                            dtype=[np.float64])
         X = check_array(X, **check_params)
         y = check_array(y, **check_params)
```

Comment:
We need to make sure `y` has the same dtype as X:

```
X = check_array(X, dtype=[np.float64, np.float32], accept_sparse=False, ensure_2d=False)
y = check_array(y, dtype=X.dtype, accept_sparse=False, ensure_2d=False)
```

Reply:
Thanks for the review, I fixed the code and added a test case.